src/class/dfu/dfu_device.c

```diff
@@ -165,22 +169,29 @@ uint16_t dfu_moded_open(uint8_t rhport, tusb_desc_interface_t const * itf_desc,
   (void) rhport;
   (void) max_len;
 
-  // Ensure this is DFU Mode
-  TU_VERIFY((itf_desc->bInterfaceSubClass == TUD_DFU_APP_SUBCLASS) &&
-            (itf_desc->bInterfaceProtocol == DFU_PROTOCOL_DFU), 0);
-
-  uint8_t const * p_desc = tu_desc_next( itf_desc );
-  uint16_t drv_len = sizeof(tusb_desc_interface_t);
+  uint16_t drv_len = 0;
+  uint8_t const * p_desc = (uint8_t*)itf_desc;
 
-  if ( TUSB_DESC_FUNCTIONAL == tu_desc_type(p_desc) )
+  while (max_len)
   {
-    tusb_desc_dfu_functional_t const *dfu_desc = (tusb_desc_dfu_functional_t const *)p_desc;
-    _dfu_state_ctx.attrs = (uint8_t)dfu_desc->bAttributes;
-
+    // Ensure this is DFU Mode
+    TU_VERIFY((((tusb_desc_interface_t const *)p_desc)->bInterfaceSubClass == TUD_DFU_APP_SUBCLASS) &&
+              (((tusb_desc_interface_t const *)p_desc)->bInterfaceProtocol == DFU_PROTOCOL_DFU), drv_len);
+  
+    p_desc = tu_desc_next( p_desc );
     drv_len += tu_desc_len(p_desc);
-    p_desc   = tu_desc_next(p_desc);
-  }
 
+    if ( TUSB_DESC_FUNCTIONAL == tu_desc_type(p_desc) )
+    {
+      tusb_desc_dfu_functional_t const *dfu_desc = (tusb_desc_dfu_functional_t const *)p_desc;
+      _dfu_state_ctx.attrs = (uint8_t)dfu_desc->bAttributes;
```

Comment:
- each ATL can have its own attributes, we should store it as an array with `CFG_TUD_DFU_ATL_MAX` (default = 2 or 4). 
- also we need to check to make sure the ALT is consecutive to make sure it is not 2 instances of DFU interface rather than 2 ALT of an DFU instance (hope my words make sense).